### PR TITLE
Adding new supported states to LACP resource module 

### PR DESCRIPTION
**Pull request by justjais**
##### SUMMARY

Adding new supported states to LACP resource module 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ios_lacp

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/module_utils/network/ios/config/lacp/lacp.py [hunk 1]
```diff
@@ -67,19 +67,40 @@ def execute_module(self):
...
+            result["rendered"] = commands
+        elif self.state == "parsed":
+            running_config = self._module.params["running_config"]
+            if not running_config:
+                self._module.fail_json(
+                    msg="value of running_config parameter must not be empty for state parsed"
+                )
+            result["parsed"] = self.get_lacp_facts(data=running_config)
+        else:
+            changed_lacp_facts = []
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/ios/config/lacp/lacp.py:96]: should be dict

File: plugins/module_utils/network/ios/config/lacp/lacp.py [hunk 2]
```diff
@@ -67,19 +67,40 @@ def execute_module(self):
         commands = list()
         warnings = list()
 
-        existing_lacp_facts = self.get_lacp_facts()
-        commands.extend(self.set_config(existing_lacp_facts))
-        if commands:
+        if self.state in self.ACTION_STATES:
+            existing_lacp_facts = self.get_lacp_facts()
+        else:
+            existing_lacp_facts = []
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/ios/config/lacp/lacp.py:73]: should be dict

File: plugins/module_utils/network/ios/config/lacp/lacp.py [hunk 3]
```diff
@@ -42,14 +42,14 @@ class Lacp(ConfigBase):
...
 
         :rtype: A dictionary
         :returns: The current configuration as a dictionary
         """
         facts, _warnings = Facts(self._module).get_facts(
-            self.gather_subset, self.gather_network_resources
+            self.gather_subset, self.gather_network_resources, data=data
         )
         lacp_facts = facts["ansible_network_resources"].get("lacp")
         if not lacp_facts:
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/ios/config/lacp/lacp.py:55]: should return dict type object

File: plugins/modules/ios_lacp.py [hunk 4]
```diff
@@ -21,10 +21,13 @@
 from __future__ import absolute_import, division, print_function
 
 __metaclass__ = type
+
 ANSIBLE_METADATA = {"metadata_version": "1.1", "supported_by": "Ansible"}
+
 DOCUMENTATION = """
 module: ios_lacp
 short_description: LACP resource module
+  IOS devices.
```
Comment (rohitthakur2590) [inline plugins/modules/ios_lacp.py:30]: is this intended ?

File: plugins/modules/ios_lacp.py [hunk 5]
```diff
@@ -47,6 +50,14 @@
             - Refer to vendor documentation for valid values.
             type: int
             required: true
+  running_config:
```
Comment (rohitthakur2590) [inline plugins/modules/ios_lacp.py:53]: this needs to be updated as per IOS device.

**Comment by justjais** [comment]
@rohitthakur2590 thanks for the review, and have implemented the suggested changes.

**Comment by justjais** [comment]
recheck

**Comment by justjais** [comment]
recheck

**Comment by justjais** [comment]
recheck

**Pull request merged**